### PR TITLE
Bugfix 2503

### LINKED ISSUE #2503: Bug when filtering variants by MT chromosome
Whoops! 

```
Traceback (most recent call last):
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/chiararasi/Documents/work/GITs/scout/scout/server/utils.py", line 29, in decorated_function
    return render_template(template_name, **context)
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/flask/templating.py", line 137, in render_template
    return _render(
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/flask/templating.py", line 120, in _render
    rv = template.render(context)
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/jinja2/environment.py", line 1090, in render
    self.environment.handle_exception()
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/Users/chiararasi/Documents/work/GITs/scout/scout/server/blueprints/variants/templates/variants/variants.html", line 159, in top-level template code
    data-content="{{ svs_table(institute, case, variant.overlapping[:20]) }}">Overlapping SVs</a>
  File "/Users/chiararasi/Documents/work/GITs/scout/scout/server/templates/layout_bs4.html", line 1, in top-level template code
    {% extends "bootstrap4.html" %}
  File "/Users/chiararasi/Documents/work/GITs/scout/scout/server/templates/bootstrap4.html", line 20, in top-level template code
    {% block body %}
  File "/Users/chiararasi/Documents/work/GITs/scout/scout/server/templates/bootstrap4.html", line 23, in block "body"
    {% block content %}
  File "/Users/chiararasi/Documents/work/GITs/scout/scout/server/templates/layout_bs4.html", line 82, in block "content"
    {% block content_main %}{% endblock %}
  File "/Users/chiararasi/Documents/work/GITs/scout/scout/server/blueprints/variants/templates/variants/variants.html", line 86, in block "content_main"
    {{ mark_heteroplasmic_mt(case.individuals, variant.samples) }}
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/jinja2/runtime.py", line 679, in _invoke
    rv = self._func(*arguments)
  File "/Users/chiararasi/Documents/work/GITs/scout/scout/server/blueprints/variants/templates/variants/utils.html", line 7, in template
    {% if gt.alt_frequency < 0.9 and gt.alt_frequency != -1 %}
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'alt_frequency'
```

### PR DESCRIPTION
**Pull request by northwestwitch**
Current master contains a bug that caused the page to crash when filtering by SNV variants that don't contain the key `alt_frequency`

Fix #2503 

**How to test -> locally**:
1. From master branch, load demo variants with the command:  `scout setup demo`
2. start the app (from master branch): `scout --demo serve`
3. Go to variantS page (SNV variants) and filter by chromosome (MT chromosome)
4. Notice that the page crashes
5. Switch to this branch and repeat the search, the page should NOT crash

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by
- [ ] tests executed by

### CODE REVIEW

**Comment by northwestwitch** [comment]
@bjhall we missed this one! Do you have time for a review?

File: scout/server/blueprints/variants/templates/variants/utils.html [hunk 1]
```diff
@@ -4,7 +4,7 @@
 {% macro mark_heteroplasmic_mt(individuals, samples) %}
   {% for ind in individuals if ind.phenotype == 2 %}
     {% for gt in samples|selectattr("sample_id", "equalto", ind.individual_id) %}
-      {% if gt.alt_frequency < 0.9 and gt.alt_frequency != -1 %}
+      {% if gt.alt_frequency and gt.alt_frequency < 0.9 and gt.alt_frequency != -1 %}
```
Comment (dnil) [inline scout/server/blueprints/variants/templates/variants/utils.html:7]: 👍

**Review by dnil — APPROVED** [review]
I couldn't reproduce the issue with the current demo? I'm all for testing for existence first though, as it could clearly be missing. Works just fine. The heteroplasmy badge could also ride in the in inheritance model column since the chr one is a little narrow, but it's all good.

**Comment by northwestwitch** [comment]
> I couldn't reproduce the issue with the current demo? I'm all for testing for existence first though, as it could clearly be missing. Works just fine. The heteroplasmy badge could also ride in the in inheritance model column since the chr one is a little narrow, but it's all good.

Did you run a `scout setup demo` command using this branch instead of master perhaps?

**Comment by dnil** [comment]
> Did you run a `scout setup demo` command using this branch instead of master perhaps?

Nope. 😅

**Comment by northwestwitch** [comment]
> Nope.

Ok but then I don't know how I ended up in the bug, but better to be safe than sorry

**Comment by dnil** [comment]
Hmm.. The only thing I can think of is that I may have tested the badge part with a previous local running (older other branch) instance on the web side after pulling master and running a new setup from cli. But again, you managed to trigger the issue, and this certainly should fix it, so all is great! 😸

**Pull request merged**